### PR TITLE
Allow safe KApply #Equals KApply in ml_pred_to_bool

### PR DESCRIPTION
**Pull request by anvacaru**
Closes https://github.com/runtimeverification/pyk/issues/552. Should be reverted if https://github.com/runtimeverification/hs-backend-booster/pull/155 gets merged.

### CODE REVIEW

**Comment by ehildenb** [comment]
In the case of #552, it seems like it's actually a safe conversion from ML predicate to bool? The reason it's safe is because all the variables in the term are element variables (not set variables), and the symbols present are all functions/contructors. So I think all you really need to check is that all the variables are element variables (not prefixed with `@`).

So rather than making `unsafe` true by default, we could just extend the algorithm to handle this case in the safe case?

We should also add tests of this particular conversion, always add tests. See https://github.com/runtimeverification/pyk/blob/949163ea8cc7f168490c84a7daa5df5481d1af9c/src/tests/unit/kast/test_manip.py#L140

**Comment by anvacaru** [comment]
@ehildenb I've added the scenario and included the simplified test case from #552.

**Review by geo2a — APPROVED** [review]
Don't know if you still need feedback, but this looks good to me! Seeing that Everett's comments are addressed, I'm approving.

File: src/pyk/kast/manip.py [hunk 1]
```diff
@@ -74,6 +74,13 @@ def bool_to_ml_pred(kast: KInner) -> KInner:
 
 
 def ml_pred_to_bool(kast: KInner, unsafe: bool = False) -> KInner:
+    def _are_all_element_vars(kast_term: KInner) -> bool:
+        if type(kast_term) == KVariable and kast_term.name.startswith('#'):
+            return False
+        if hasattr(kast_term, 'args'):
+            return all(_are_all_element_vars(arg) for arg in kast_term.args)
+        return True
+
```
[the remark this answers is not in the document]
Comment (anvacaru) [reply src/pyk/kast/manip.py:83]: Addressed this in #616.

**Pull request merged**